### ISSUE TITLE
Ignore dependencies which are a workspace member

### ISSUE THREAD

**Issue by jayvdb** [trailofbits/cargo-unmaintained#277]
In our workspace Cargo.toml we have each member listed in `[dependencies]` as `memberA = { path = "..." }`

Then memberB's Cargo.toml it includes `memberA.workspace = true`

And cargo-unmaintained lists them as `(no repository)`.

**Comment by smoelius** [comment]
Are the workspace members published to crates.io?

**Comment by jayvdb** [comment]
no, they are not published to any registry.

They are specified by `path = ` only.

**Comment by smoelius** [comment]
Sorry, but I don't quite understand the issue. Could you share the warning that is produced, possibly redacted?

Also, in your description, I don't understand the relationship between `memberA` and `memberB`. Is the following really what you meant to write?

> Then memberB's Cargo.toml it includes `memberA.workspace = true`

I don't understand what it means for memberB's Cargo.toml to include that syntax. Could you maybe share an actual snippet from memberB's Cargo.toml, again possibly redacted?

Part of my confusion stems from the fact that dependencies specified by path appear to already be ignored: https://github.com/trailofbits/cargo-unmaintained/blob/9b39e320b263910b2a4dc57f0fe6dd6027d7f6fd/src/main.rs#L639-L642

**Comment by jayvdb** [comment]
Yes, I meant to write `memberA.workspace = true` which is the same as `memberA = { workspace = true }`.

top level `Cargo.toml`:

```toml
[workspace.dependences]
memberA = { path = "path/to/memberA" }
memberB = { path = "path/to/memberB" }

[workspace]
members = [
  "path/to/memberA",
  "path/to/memberB",
]
```

`path/to/memberB/Cargo.toml`:

```toml
[dependencies]
memberA.workspace = true
```

`memberB` depends on `memberA`, which is described in the top level `Cargo.toml`.

**Comment by smoelius** [comment]
Thank you. So `memberA` or `memberB` (or both) is reported as unmaintained? Do they legitimately meet one of the criteria [in the README](https://github.com/trailofbits/cargo-unmaintained?tab=readme-ov-file#cargo-unmaintained)?

**Comment by jayvdb** [comment]
`memberA` is reported as "memberA (no repository)" followed by an indented subset of its dependencies which are not the latest version.

Oh I think I can guess what is happening.  The entire workspace does not define a `repository = ..` in any `Cargo.toml` , as none of the members are published to a registry.

So `memberA` does not have a repository, but IMO that should not trigger this tool saying it is unmaintained, as `memberA` is always referring to using `path = ...`.  Saying that a workspace member is unmaintained is weird.

**Comment by smoelius** [comment]
> Saying that a workspace member is unmaintained is weird.

I think this should be fixed in the latest release ([1.1.0](https://github.com/trailofbits/cargo-unmaintained/releases/tag/v1.1.0)) if you want to try it out.

**Comment by jayvdb** [comment]
That works well. Thanks.
